### PR TITLE
Peer heap

### PR DESCRIPTION
**Pull request by nomis52**
Depends on #144

### CODE REVIEW

**Comment by prashantv** [comment]
lgtm, you probably want to merge the other changes and rebase before merging this.

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,12 @@ TEST_PKGS := $(addprefix github.com/uber/tchannel-go,$(TEST_PKGS_NOPREFIX))
 BUILD := ./build
 SRCS := $(foreach pkg,$(PKGS),$(wildcard $(pkg)/*.go))
 export GOPATH = $(GODEPS):$(OLDGOPATH)
-export PATH := $(realpath ./scripts/travis/thrift-release/linux-x86_64):$(PATH)
+
+PLATFORM := $(shell uname -s | tr '[:upper:]' '[:lower:]')
+ARCH := $(shell uname -m)
+THRIFT_REL := ./scripts/travis/thrift-release/$(PLATFORM)-$(ARCH)
```
Comment (breerly) [inline Makefile]: This is sweet.

**Comment by akshayjshah** [comment]
@nomis52 Ping :)

Can I help get this ready to land?

**Comment by prashantv** [comment]
As of #269, we've documented that the peer heap is not safe for concurrent use, and expects callers (in this case `PeerList`) to serialize access to the heap. We no longer use atomic operations in the heap, so this PR shouldn't be required anymore.

**Pull request closed without merging**